http-clients/netty-nio-client/src/test/java/software/amazon/awssdk/http/nio/netty/fault/ServerNotRespondingTest.java

```diff
@@ -114,12 +114,10 @@ public void connectionNotRespond_doesNotAckPing_shouldNotReuse() throws Exceptio
         assertThatThrownBy(() -> firstRequest.join()).hasCauseInstanceOf(IOException.class);
 
         // The second request should pick up a new healthy channel - Channel 2
-        CompletableFuture<Void> secondRequest = sendGetRequest();
+        sendGetRequest().join();
 
         // The third request should reuse the previous channel - Channel 2
-        CompletableFuture<Void> thirdRequest = sendGetRequest();
-
-        CompletableFuture.allOf(secondRequest, thirdRequest).join();
+        sendGetRequest().join();
```

Comment:
Yes.

In the old implementation, the pool will create new streams on a connection that hasn't been fully initialized yet, up to a limit negotiated once when the client was created.

In the new implementation, the pool will only create one new stream on a connection until the protocol has been negotiated, at which time it can have more streams created on it up to the negotiated client/server-configured stream limit.

In the old implementation, if the service changes the MAX_CONCURRENT_CONNECTIONS, the client has to bounce their service to get the newest value. At best, this is under-utilizing a connection (because the limit increased). At worst, we keep trying to create new streams on a connection that is maxed out and we fail every request (because the limit decreased).